### DISCUSSION TITLE
Multidimensional input for segmentation of 2D images

### DISCUSSION THREAD

**Discussion by akraskovSMDB** [ivadomed/ivadomed#1300] [General]
Hi, 

I started this discussion in axondeepseg (https://github.com/axondeepseg/axondeepseg/discussions/745#discussioncomment-6254591) but was advised to raise it here.

Is it possible to use multidimensional input for model training?

For example, can I use all three channels of RGB colour images for training (instead of single intensity value per pixel as it is done now when RGB is converted into grey automatically)?

@mathieuboudreau mentioned that  IVADOMED also trains on MRI data, suggesting that it can handle 3D data but I don't think this is what I want. As far as I understand for 3D MRI the date is treated as a 3D voxel, i.e. there are smoothing operations applied to a 3D volume.

I want to treat each individual data colour channel independently as an additional source of information about a pixel within 2d images rather than a third dimension of the 2d image. In some sense 3 is not a limit, I might have more information about each pixel (e.g. different stainings).

To summarise, I am asking whether a model for the segmentation of a 2D image can be trained using a multidimensional property vector for each 2D pixel.

Can this be done with ivadomed? 
Can this be done using U-net framework? If the answers are no and yes, could you advise a different code for doing this?

Thank you for your help
